app/src/main/java/io/reark/rxgithubapp/network/NetworkModule.java

```diff
@@ -26,36 +26,31 @@
 package io.reark.rxgithubapp.network;
 
 import com.google.gson.Gson;
-import com.squareup.okhttp.OkHttpClient;
+
+import net.simonvt.schematic.annotation.NotNull;
 
 import javax.inject.Named;
 import javax.inject.Singleton;
 
 import dagger.Module;
 import dagger.Provides;
-import retrofit.client.Client;
-import retrofit.client.OkClient;
+import okhttp3.OkHttpClient;
 
 @Module
 public final class NetworkModule {
 
     @Provides
     @Singleton
-    public NetworkApi provideNetworkApi(@Named("okClient") Client client) {
+    public NetworkApi provideNetworkApi(OkHttpClient client) {
         return new NetworkApi(client);
     }
 
-    @Provides
-    @Singleton
-    @Named("okClient")
-    public Client provideOkClient(OkHttpClient okHttpClient) {
-        return new OkClient(okHttpClient);
-    }
 
     @Provides
     @Singleton
     public OkHttpClient provideOkHttpClient(NetworkInstrumentation<OkHttpClient> networkInstrumentation) {
-        return networkInstrumentation.decorateNetwork(new OkHttpClient());
+        OkHttpClient client = new OkHttpClient().newBuilder().addNetworkInterceptor(networkInstrumentation.getInterceptor()).build();
+        return networkInstrumentation.decorateNetwork(client);
```

Comment:
https://square.github.io/okhttp/3.x/okhttp/okhttp3/class-use/Interceptor.html

Interceptor it is immutable list, so I added in builder